runtime/tests/redirector/nativevmargs/testnativevmargs.c

```diff
@@ -0,0 +1,182 @@
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file testnativevmargs.c
+ * Executable provides a native framework to create a VM with specified VM arguments.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "jni.h"
+
+#if defined(WIN32)
+#include <windows.h>
+#elif defined(LINUX) || defined(AIXPPC) || defined(OSX)
+#include <dlfcn.h>
+#else
+#include <dll.h>
+#include "atoe.h"
+#endif
+
+#define J9PATH_LENGTH_MAXIMUM 1024
+#if defined(WIN32)
+#define J9PATH_DIRECTORY_SEPARATOR '\\'
+#define J9PATH_JVM_LIBRARY "jvm.dll"
+#else /* defined(WIN32) */
+#define J9PATH_DIRECTORY_SEPARATOR '/'
+#if defined(OSX)
+#define J9PATH_JVM_LIBRARY "libjvm.dylib"
+#else /* defined(OSX) */
+#define J9PATH_JVM_LIBRARY "libjvm.so"
+#endif /* defined(OSX) */
+#endif /* defined(WIN32) */
+
+/* The JVM launcher. */
+int main(int argc, char **argv)
+{
+	JavaVMInitArgs vm_args;
+	JavaVMOption * args = NULL;
+	JavaVM * vm = NULL;
+	JNIEnv * env = NULL;
+	jint result;
+	jboolean ignoreUnrecognized = JNI_FALSE;
+	int jvmOptionCount = 0;
+	int optionCount = 0;
+	char * jvmPath = NULL;
+	int rc = 0;
+	char buffer[J9PATH_LENGTH_MAXIMUM] = {0};
+	typedef jint (JNICALL * createJVMFP)(JavaVM ** pvm, void ** penv, void * vm_args);
+	createJVMFP createJavaVM = NULL;
+#if defined(WIN32)
+	HINSTANCE handle = NULL;
+#else
+	void * handle = NULL;
+#endif
+
+	fprintf(stdout, "[MSG] Starting up native VM options test.\n");
+	if (argc < 2) {
+		fprintf(stderr, "[ERR] Insufficient arguments passed to test.\n");
+		fprintf(stderr, "[MSG] Usage: nativevmargs <jvmlibpath> <options>\n");
+		rc = 1;
+		goto fail;
+	}
+	fflush(stdout);
+
+	jvmPath = argv[1];
+	optionCount = argc - 2;
+
+	args = malloc(sizeof(JavaVMOption) * optionCount);
+	if (!args) {
+		fprintf(stderr, "[ERR] Failed memory allocation for JVM arguments.\n");
+		rc = 2;
+		goto fail;
+	}
+
+	/* Create the JVM options */
+	for (int i = 2; jvmOptionCount < optionCount; i++) {
+		if (0 == strncmp(argv[i], "-Custom:", 8)) {
+			/* convert custom options to JVM options */
+			if (0 == strcmp(argv[i], "-Custom:WhitespaceOption")) {
```

Comment:
Do we still need to test with a single space now that MultipleWhitespaceOption is added? Running command line tests are a little costly, more so on certain platforms, since the JVM needs to start up each time.

Reply:
MultipleWhitespaceOption will cover the single space case - will be removing single space tests